pkg/cloud/azureprovider.go

```diff
@@ -740,10 +839,55 @@ func (az *Azure) AllNodePricing() (interface{}, error) {
 func (az *Azure) NodePricing(key Key) (*Node, error) {
 	az.DownloadPricingDataLock.RLock()
 	defer az.DownloadPricingDataLock.RUnlock()
+
 	azKey, ok := key.(*azureKey)
 	if !ok {
 		return nil, fmt.Errorf("azure: NodePricing: key is of type %T", key)
 	}
+	config, _ := az.GetConfig()
+	if config.SpotLabel != "" && config.SpotLabelValue != "" {
+		labelsMapStr := fmt.Sprintf("%s", azKey)
+		spotLabelMapStr := fmt.Sprintf(" %s:%s ", config.SpotLabel, config.SpotLabelValue)
+		if strings.Contains(labelsMapStr, spotLabelMapStr) {
```

Comment:
I'm a little lost here. Does `azKey.String()` (which I think is what gets called when you `%s` format it?) just output space-separated labels of the node?

If that assumption is right, does it always have a leading/trailing space? This is the same kind of question with the placement of `ondemand` in the `ReplaceAll`, I just want to make sure we're robust to the placement of the specific label within the list.

Reply:
Here is @Subreptivus on the topic
> Padding is there, because labelsMapStr will have all the labels separated by space, and substr can find needed string even in the middle of another string (this is kind of just for safety, because you can set spot labels while deploying cluster and it could be something really simple, which could be already there within another label, like: kubernetes.azure.com/scalesetpriority: mylabel and kubernetes.azure.com/scalesetpriority: mylabelspot).

however maybe doing a split on " " and searching for the unpadded string in the resulting slice would be more robust

Reply:
Hmm yeah this kind of string manipulation just doesn't feel right to me. I'm particularly not a fan of the `fmt.Sprintf("%s", azKey)`, which seems to be relying on behavior of `azKey.String()` (I think?) that I wouldn't expect.